backend-db/Rhyolite/Backend/DB.hs

```diff
@@ -65,27 +66,55 @@ import Rhyolite.Schema
 
 type Db m = (PersistBackend m, PostgresRaw m, SqlDb (PhantomDb m))
 
+runDbPersistTransactionMode
+  :: (MonadIO m, MonadLoggerIO m)
+  => Pg.IsolationLevel
+  -> Pg.ReadWriteMode
+  -> Pool Pg.Connection
+  -> DbPersist Postgresql (LoggingT IO) a
+  -> m a
+runDbPersistTransactionMode isoLevel rwMode dbPool (DbPersist act) = do
+  logger <- askLoggerIO
+  liftIO $ withResource dbPool $ \conn ->
+    Pg.withTransactionMode (Pg.TransactionMode isoLevel rwMode) conn $
+      runLoggingT (runReaderT act (coerce conn)) logger
+
 -- | Runs a database action using a given pool of database connections
 -- The 'f' parameter can be used to represent additional information about
 -- how/where to run the database action (e.g., in a particular schema).
 -- See instances below.
 class RunDb f where
+  -- | Runs a transaction at serializable isolation level and automatically retries
+  --   in case of a serialization error.
   runDb :: (MonadIO m, MonadLoggerIO m, Coercible conn Pg.Connection)
         => f (Pool conn)
         -> Serializable a
         -> m a
 
+  -- | Runs a read-only transaction at repeatable read isolation level.
+  --   No retrying is done.
+  --   NB: "Read-only" is not statically checked!
+  runDbReadOnlyRepeatableRead
```

Comment:
It's probably worth adding a comment here to explain why this *particular* combination of transaction parameters is "blessed" in `RunDb`: Because it's the only transaction configuration that cannot fail due to serialization error (therefore does not need retry) and won't wait unbounded time (like `SERIALIZABLE READ ONLY DEFERRED` can).